olp-cpp-sdk-core/src/client/OlpClient.cpp

```diff
@@ -324,6 +319,28 @@ void OlpClient::OlpClientImpl::SetSettings(const OlpClientSettings& settings) {
   settings_ = settings;
 }
 
+void OlpClient::OlpClientImpl::AddBearer(const std::string& path,
+                                         bool query_empty,
+                                         http::NetworkRequest& request) const {
+  auto settings = settings_.authentication_settings;
```

Comment:
No need to copy. Be aware that boost::optional is not a shared_ptr. it actually does a deep copy internally.